### PR TITLE
#163 Fixing Error when executing the hook under AEM 6.3

### PR DESCRIPTION
**Pull request by karolisma**
Added newer sling api version.

### CODE REVIEW

File: accesscontroltool-bundle/src/main/java/biz/netcentric/cq/tools/actool/session/impl/SessionManagerImpl.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        if(session != null) {
+            return session;
+        }
+
+        return getServiceUserSession(session);
+    }
+
+    @Override
+    public void close(Session session) {
+        if(session.equals(sessionHolder.get())) {
```
Comment (santiagozky) [inline accesscontroltool-bundle/src/main/java/biz/netcentric/cq/tools/actool/session/impl/SessionManagerImpl.java]: maybe instead we should wrap the session in a custom implementation that prevents closing it? that way a developer cannot accidentally just call session.logout()

**Comment by kwin** [comment]
I would very much be in favor of passing a session to be used as parameter. That way the caller (i.e. JMX or Vault Hook) can explicitly determine which session/service user to use. Hiding that in a kind of magic service (using either service resource resolvers or thread resource resolvers) is too magic for me. Also it is not clear to me, how would close the session/resource resolver then?

We should make sure, that the session is closed as soon as possible (and not being reused among multiple threads)

**Comment by ghenzler** [comment]
I agree with @kwin that we should reuse credentials (but not sessions) as much as possible. This means for me
- For the install hook case, the package manager session should be cloned for all spots where currently the admin session is opened https://java.net/jira/browse/JSR_333-27
- For the plain JMX case (not via /system/console/jmx) there is definitely now session we can rely on. For that we probably just have to create a new session 
- For the System Console JMX case, it is probably wise to use the sling request session for cloning (@kwin WDYT? better than creating a new session with a system user configured IMHO)

Generally I would not like to change anything regarding session lifetimes and session count ATM - I would try to clone sessions from existing sessions if possible (and only create from scratch if really necessary)

**Comment by kwin** [comment]
> For the System Console JMX case, it is probably wise to use the sling request session for cloning (@kwin WDYT? better than creating a new session with a system user configured IMHO)

JMX has never access to a Sling Request. Even if accessed through the Felix Web Console, this request is not handled by Sling, therefore only a HttpServletRequest is available which has no `ResourceResolver` or `Session`.

**Comment by ghenzler** [comment]
@kwin You don't have access to the Sling Request, but that is not needed. I could image that [1] does work (albeit not a very clean solution) - @karolisma have you tested if you actually can get a session like this? (and if yes, does it still work if you use plain curl to trigger it?)

[1]
https://github.com/karolisma/accesscontroltool/blob/9c0f02ee9682a09019e6d253704ae641ebac8176/accesscontroltool-bundle/src/main/java/biz/netcentric/cq/tools/actool/session/impl/SessionManagerImpl.java#L70

**Comment by kwin** [comment]
> I could image that [1] does work (albeit not a very clean solution) 

`resourceResolverFactory.getThreadResourceResolver();` returns null for Non-Sling threads/requests.

**Comment by karolisma** [comment]
@ghenzler I have tested, it works on installer hook and via jmx (/system/console), if user logged in has permissions needed.
I haven't tried it via curl.

**Pull request closed without merging**